src/H5FDsubfiling/H5FDioc_threads.c

```diff
@@ -362,50 +334,44 @@ ioc_main(ioc_data_t *ioc_data)
         /* Probe for incoming work requests */
         if (MPI_SUCCESS !=
             (mpi_code = (MPI_Iprobe(MPI_ANY_SOURCE, MPI_ANY_TAG, context->sf_msg_comm, &flag, &status))))
-            H5_SUBFILING_MPI_GOTO_ERROR(-1, "MPI_Iprobe failed", mpi_code);
+            HGOTO_DONE((hg_thread_ret_t)FAIL);
 
         if (flag) {
-            double queue_start_time;
-            int    count;
-            int    source = status.MPI_SOURCE;
-            int    tag    = status.MPI_TAG;
+            int count;
+            int source = status.MPI_SOURCE;
+            int tag    = status.MPI_TAG;
 
-            if ((tag != READ_INDEP) && (tag != WRITE_INDEP) && (tag != TRUNC_OP) && (tag != GET_EOF_OP))
-                H5_SUBFILING_GOTO_ERROR(H5E_ARGS, H5E_BADVALUE, -1, "invalid work request operation (%d)",
-                                        tag);
+            if (tag != READ_INDEP && tag != WRITE_INDEP && tag != TRUNC_OP && tag != GET_EOF_OP)
+                HGOTO_DONE((hg_thread_ret_t)FAIL);
```

Comment:
As a bit of irony, this seems like a very good example where separate error stacks would be nice so that when an error occurs in this thread, it can return an ID for it's error stack and the process can merge that stack with its own so we keep errors like this rather than getting an unhelpful FAIL.

Reply:
I was thinking about that too while I was working on these background thread errors.  :-)

I'll add an item in my TODO list to work on how to make that happen.